src/java.base/share/classes/java/io/PipedOutputStream.java

```diff
@@ -163,6 +163,7 @@ public void write(byte[] b, int off, int len) throws IOException {
      */
     @Override
     public synchronized void flush() throws IOException {
+        PipedInputStream sink = this.sink;
```

Comment:
```suggestion
        var sink = this.sink;
```
As seen in other methods.

Reply:
On second thought, this is probably not necessary; write to the sink field is in another synchronized method, and this method is synchronized already. Is the goal here to remove the synchronized on flush?

Reply:
I think it's better to keep `synchronized` over the method this far